support/metrics/metricsTracker.go

```diff
@@ -286,3 +319,28 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func (mt *MetricsTracker) mergeEventProps(eventProps interface{}) (map[string]interface{}, error) {
+	var m map[string]interface{}
+	eventJSON, e := json.Marshal(eventProps)
+	if e != nil {
+		return nil, fmt.Errorf("metric - could not marshal event props json: %s", e)
+	}
+
+	e = json.Unmarshal(eventJSON, &m)
```

Comment:
marshalling and unmarshalling so many times seems excessive and expensive -- there must be a better way of converting from a `struct` to `map[string]interface{}`.

wonder if this library can help (we already depend on it in `ccxt.go`): [github.com/mitchellh/mapstructure](github.com/mitchellh/mapstructure)